### PR TITLE
Working Casio Loopy driver

### PR DESCRIPTION
**Pull request by philipjbennett**
Systems promoted to working
---------------------------
Casio Loopy [Phil Bennett, Ryan Holtz]

New working software list items
-------------------------------
Chakra-kun no Omajinai Paradise [Rachel Simone Weil]

### CODE REVIEW

File: src/devices/bus/casloopy/slot.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	virtual uint8_t read_ram(offs_t offset) { return 0xff; }
+	virtual void write_ram(offs_t offset, u8 data) { }
+
+	void rom_alloc(uint32_t size);
+	void nvram_alloc(uint32_t size);
+	uint16_t* get_rom_base() const { return m_rom; }
+	uint32_t get_rom_size() const { return m_rom_size; }
+
+	uint8_t* get_nvram_base() { return &m_nvram[0]; }
+	uint32_t get_nvram_size() { return m_nvram.size(); }
```
[the remark this answers is not in the document]
Comment (MooglyGuy) [reply src/devices/bus/casloopy/slot.h:33]: Please point explicitly at the bus device which best represents the cartridge in this case. Happy to work on revising it, but only if I have a template to work off of.

File: src/mame/casio/casloopy_tbl.ipp
[the remark this answers is not in the document]
Comment (MooglyGuy) [reply src/mame/casio/casloopy_tbl.ipp:1]: There's no obvious derivation for the tables at this point. If someone else wants to identify common sections and reduce the data load, more power to them, but for now these tables represent months of work with a logic analyzer.

File: src/devices/bus/casloopy/slot.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (cuavas) [reply src/devices/bus/casloopy/slot.h:33]: Virtual Boy is pretty simple.
* In the slot’s `call_load`, allocate regions if loading loose software: https://github.com/mamedev/mame/blob/mame0252/src/devices/bus/vboy/slot.cpp#L52
* Then in the cartridge, you can find the regions relative to the parent device like `"^rom"` etc.: https://github.com/mamedev/mame/blob/mame0252/src/devices/bus/vboy/rom.cpp#L42

File: src/mame/casio/casloopy_tbl.ipp
[the remark this answers is not in the document]
Comment (philipjbennett) [reply src/mame/casio/casloopy_tbl.ipp:1]: I can simplify 2 or maybe even 3 of them. I'm not sure about s_m0_pri_6.

[the remark this answers is not in the document]
Comment (cuavas) [reply src/mame/casio/casloopy_tbl.ipp:1]: The tables aren’t a deal breaker.  Don’t stress over that.  The `save_pointer` stuff is a heap smash, the cartridge thing is something we’re really trying to get away from to reduce the number of bad examples to copy, and the rest of the comments are pretty trivial things to address.

**Pull request merged**